[Asp.Net Core, HttpClient] Clarify Enrich callback sequence (#2011)
open-telemetry/opentelemetry-dotnet-contrib: src/OpenTelemetry.Instrumentation.AspNetCore/README.md

The `Enrich` option is specific to this instrumentation, and is provided to
get access to `HttpRequest` and `HttpResponse`.

When overriding the default settings provided by instrumentation or adding
additional telemetry, it is important to consider the sequence of callbacks.
Generally, it is recommended to use `EnrichWithHttpResponse` for any activity
enrichment that does not need access to exceptions, as the instrumentation

> library sets all defaults before this callback. The following is the sequence in

Reviewer: ```suggestion
library populates telemetry using OTel sem conv, before this callback. The following is the sequence in
```

which these callbacks are executed:

1) Processor `OnStart`
2) `EnrichWithHttpRequest`
3) `EnrichWithException`
4) `EnrichWithHttpResponse`

> 5) Processor `OnEnd`

Reviewer: nit: Would be nice to call out that `SetTag` will override any existing tag of same name, and `SetTag` with null is a way to achieve removing a Tag completely.

Author: I added in the example code.

The `Enrich` option is specific to this instrumentation, and is provided to
get access to `HttpRequest` and `HttpResponse`.

When overriding the default settings provided by instrumentation or adding
additional telemetry, it is important to consider the sequence of callbacks.
Generally, it is recommended to use `EnrichWithHttpResponse` for any activity
enrichment that does not need access to exceptions, as the instrumentation

> library populates all telemetry following the [OTel
> specification](https://github.com/open-telemetry/semantic-conventions/tree/main/docs/http)

Reviewer 2: ```suggestion
library populates all telemetry following the [OpenTelemetry semantic
convention](https://github.com/open-telemetry/semantic-conventions/tree/v1.27.0/docs/http)
```

Same for the second readme

Author: Thanks - updated

before this callback. The following is the sequence in which these callbacks are
executed:

1) Processor `OnStart`
2) `EnrichWithHttpRequest`
3) `EnrichWithException`
4) `EnrichWithHttpResponse`
5) Processor `OnEnd`

As an example, if you need to override the default DisplayName or tags set by
the library you can do so as follows:

```csharp
.AddAspNetCoreInstrumentation(o =>
{
  o.EnrichWithHttpResponse = (activity, response) =>
  {
      // Access request object if needed
      // response.HttpContext.Request

>       activity.DisplayName = CustomDisplayName;

Reviewer 3: nit
```suggestion
      activity.DisplayName = "CustomDisplayName";
```
